src/components/SearchField.js

```diff
@@ -0,0 +1,276 @@
+        inputElement.blur()
+      }
+    },
+    onInputValueChange: ({ inputValue, type }) => {
+      // Clear the selection when the search field is cleared via backspace
+      if (inputValue === '' && type === useCombobox.stateChangeTypes.InputChange) {
+        // Clear the redux state tracking the selected feature and let this
+        // component rerender based on that rather than calling
+        // selectItem(null) here
+        setSelectedFeature({ selectedItem: null })
+      // If inputValue changes while the search menu is closed, it's because 
+      // another component change selectedFeature, and that should be
+      // reflected in the selectedItem of the search field.
+      } else if (!isOpen && inputValue) {
+        const definiteMatch = localItems.find(item => item.primary === inputValue)
+        if (definiteMatch) {
+          selectItem(definiteMatch)
+        }
+      }
+    },
+    onSelectedItemChange: setSelectedFeature,
+    onHighlightedIndexChange: ({ highlightedIndex, type }) => {
+      switch (type) {
+        // Ignore the hidden highlight change which occurs when selecting an
+        // item and the menu closes.
+        case useCombobox.stateChangeTypes.ItemClick:
+        case useCombobox.stateChangeTypes.InputKeyDownEnter:
+        case useCombobox.stateChangeTypes.FunctionSelectItem:
+          break
+        default:
+          setHighlightedItem(inputItems[highlightedIndex])
+      }
+    },
+    stateReducer: (state, actionAndChanges) => {
+      const {type, changes} = actionAndChanges
+      switch (type) {
+        // Don't select whatever happened to be highlighted if the user
+        // switches to another window.
+        case useCombobox.stateChangeTypes.InputBlur:
+          return {
+            ...changes,
+            inputValue: state.inputValue,
+            selectedItem: state.selectedItem,
+          }
+        default:
+          return changes
+      }
+    },
+  })
+
+  useEffect(() => {
+    if (typeof setHighlightedFeature === "function") {
+      setHighlightedFeature({highlightedItem})
+    }
+  }, [highlightedItem, setHighlightedFeature])
+
+  // Set up geocoder
+  const geocodedItems = useGeocoder({
+    inputValue,
+    // Skip geocoding API call if the results won't even be seen
+    enabled: isOpen,
+    config: geocoderConfig,
+    onNewFeatures: useCallback((features) => (
+      features.map((f) => {
+        const localFeature = localItems.find(
+          (item) => (item.geometry && turf.booleanPointInPolygon(f.center, item))
+        )
+        return {
+          ...localFeature,
+          secondary: f.place_name,
+          relevance: f.relevance,
+        }
+      // Filter out results which aren't inside any of the regions this app
+      // tracks. For example, a search for "west lak" includes Lake Mundi in
+      // Victoria, even though the geocoding request was restricted to Sourth
+      // Australia.
+      }).filter(f => f.primary)
```

Comment:
Not really sure if this really matters performance-wise but if you put this filter statement before the .map() you'll make fewer map calls (and .find() calls too) so could be good just in case

Reply:
It's actually filtering out things that failed to map. It's not obvious thanks to spreading `...localFeature`, but the `primary` key it's filtering on doesn't exist until after the map has done its thing.